packages/schemas/carbon-resources.schema.json

```diff
@@ -296,6 +376,100 @@
+                },
+                "status": {
+                  "oneOf": [
+                    {
+                      "type": "string",
+                      "default": "draft",
+                      "enum": ["deprecated", "draft", "experimental", "stable"]
+                    },
+                    {
+                      "type": "object",
+                      "properties": {
+                        "key": {
+                          "type": "string",
+                          "default": "draft",
+                          "enum": ["deprecated", "draft", "experimental", "stable"]
+                        },
+                        "note": {
+                          "type": "string",
+                          "description": "Explains the key. This is often used to communicate the remaining effort to get something from draft or experimental to stable, estimated stability dates, and alternative assets to use when the asset has been deprecated."
+                        }
+                      },
+                      "required": ["key", "note"],
+                      "additionalProperties": false
+                    }
+                  ],
+                  "description": "Used to set consumption expectations and the likelihood of future changes."
+                },
+                "tool": {
+                  "type": "string",
+                  "enum": [
+                    "adobe-ase",
+                    "adobe-xd",
+                    "axure",
+                    "figma",
+                    "invision-freehand",
+                    "sketch"
+                  ],
+                  "description": "Design kit's compatible tool."
+                },
+                "type": {
+                  "type": "string",
+                  "enum": ["elements", "guidelines", "ui", "wireframes"],
+                  "description": "Design kit primary categorization."
+                },
+                "url": {
+                  "type": "string",
+                  "description": "Design kit URL."
+                },
+                "action": {
+                  "type": "string",
+                  "default": "link",
+                  "enum": ["download", "link"],
+                  "description": "Determines the action icon."
+                },
+                "license": {
+                  "type": "string",
+                  "enum": ["apache-2.0", "ibm-internal", "mit"],
+                  "description": "Design kit license."
+                },
+                "noIndex": {
+                  "type": "boolean",
+                  "default": false,
+                  "description": "If set to true, the catalogs will exclude the kit."
+                }
+              },
+              "additionalProperties": false,
+              "oneOf": [
+                { "required": ["$ref"] },
+                { "required": ["name", "tool", "type", "status", "url"] }
+              ]
+            }
+          }
```

Comment:
Defining it as a oneOf of two separate objects will prevent the user from being able to mix `$ref` with the other properties
```suggestion
                    "patternProperties": {
                        "^.*$": {
                            "description": "Design kits contain elements, guidelines, user interface components and patterns, and wireframes for various design tools as the design counterpart to coded assets.",
                            "oneOf": [
                                {
                                    "type": "object",
                                    "properties": {
                                        "name": {
                                            "type": "string",
                                            "description": "Design kit display name (often includes the tool and color theme.) Use title-case capitalization."
                                        },
                                        "description": {
                                            "type": "string",
                                            "description": "Design kit description ideally between 50-160 characters in length. Use sentence-case capitalization."
                                        },
                                        "status": {
                                            "oneOf": [
                                                {
                                                    "type": "string",
                                                    "default": "draft",
                                                    "enum": [
                                                        "deprecated",
                                                        "draft",
                                                        "experimental",
                                                        "stable"
                                                    ]
                                                },
                                                {
                                                    "type": "object",
                                                    "properties": {
                                                        "key": {
                                                            "type": "string",
                                                            "default": "draft",
                                                            "enum": [
                                                                "deprecated",
                                                                "draft",
                                                                "experimental",
                                                                "stable"
                                                            ]
                                                        },
                                                        "note": {
                                                            "type": "string",
                                                            "description": "Explains the key. This is often used to communicate the remaining effort to get something from draft or experimental to stable, estimated stability dates, and alternative assets to use when the asset has been deprecated."
                                                        }
                                                    },
                                                    "required": [
                                                        "key",
                                                        "note"
                                                    ],
                                                    "additionalProperties": false
                                                }
                                            ],
                                            "description": "Used to set consumption expectations and the likelihood of future changes."
                                        },
                                        "tool": {
                                            "type": "string",
                                            "enum": [
                                                "adobe-ase",
                                                "adobe-xd",
                                                "axure",
                                                "figma",
                                                "invision-freehand",
                                                "sketch"
                                            ],
                                            "description": "Design kit's compatible tool."
                                        },
                                        "type": {
                                            "type": "string",
                                            "enum": [
                                                "elements",
                                                "guidelines",
                                                "ui",
                                                "wireframes"
                                            ],
                                            "description": "Design kit primary categorization."
                                        },
                                        "url": {
                                            "type": "string",
                                            "description": "Design kit URL."
                                        },
                                        "action": {
                                            "type": "string",
                                            "default": "link",
                                            "enum": [
                                                "download",
                                                "link"
                                            ],
                                            "description": "Determines the action icon."
                                        },
                                        "license": {
                                            "type": "string",
                                            "enum": [
                                                "apache-2.0",
                                                "ibm-internal",
                                                "mit"
                                            ],
                                            "description": "Design kit license."
                                        },
                                        "noIndex": {
                                            "type": "boolean",
                                            "default": false,
                                            "description": "If set to true, the catalogs will exclude the kit."
                                        }
                                    },
                                    "additionalProperties": false,
                                    "required": [
                                        "name",
                                        "tool",
                                        "type",
                                        "status",
                                        "url"
                                    ]
                                },
                                {
                                    "type": "object",
                                    "properties": {
                                        "$ref": {
                                            "type": "string",
                                            "description": "Reference to a design kit."
                                        }
                                    },
                                    "additionalProperties": false,
                                    "required": [
                                        "$ref"
                                    ]
                                }
                            ]
                        }
                    }
```